### PR TITLE
feat(graph): Show labels on hover while hidden

### LINKED ISSUE #3785: Graph2 show labels on hover (if the edges were hidden)

### PR DESCRIPTION
**Pull request by jbrundage**
Fixes #3785

Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [ ] The commit is signed.
- [x] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

## Testing:
The change can be viewed on this html page: apps/graph2/gh_3785.html

### CODE REVIEW

**Comment by jbrundage** [comment]
@GordonSmith please review

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: apps/graph2/gh_3785.html [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        right: 0px;
+        bottom: 0px;
+        left: 0px;
+    }
+    .common_IconBar{
+        display: none;
+    }
+    </style>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/Faker/3.1.0/faker.min.js"></script>
+    <!-- <script src="bostock_data.js"></script> -->
```
Comment (GordonSmith) [inline apps/graph2/gh_3785.html:48]: Remove commented code.

File: packages/graph/src/graph2/graph.ts [hunk 2]
```diff
@@ -432,16 +432,32 @@ export class Graph2 extends SVGZoomWidget {
...
             .style("filter", d => vertexMap && vertexMap[d.id] ? "url(#" + this.id() + "_glow)" : null)
-            .transition().duration(this.transitionDuration())
+            .transition().duration(context.showVertexLabelsOnHighlight() ? 0 : this.transitionDuration())
             .on("end", function (d) {
+                let _showLabel = showLabel;
                 if (vertexMap && vertexMap[d.id]) {
                     if (d.element.node() && d.element.node().parentNode) {
                         d.element.node().parentNode.appendChild(d.element.node());
                     }
+                    if(!_showLabel)_showLabel = context.showVertexLabelsOnHighlight();
```
Comment (GordonSmith) [inline packages/graph/src/graph2/graph.ts:447]: I will have a play, not sure if this should be needed?

Comment (jbrundage) [reply packages/graph/src/graph2/graph.ts:447]: That line tells it to set showLabel to true if they have `showVertexLabelsOnHighlight` enabled

**Pull request closed without merging**